### PR TITLE
Add DateTIme.Add(Duration) and DateTimeOffset.Add(Duration)

### PR DESCRIPTION
**Pull request by preeder**
Added extension class with two methods, Add(Duration), to DateTime and DateTimeOffset.  These match functionality which already exists in those classes when adjusting a moment in time by a TimeSpan.

### CODE REVIEW

File: src/Iso8601DurationHelper/Duration.cs [hunk 1]
```diff
@@ -678,4 +678,32 @@ private enum DurationComponent
             Seconds = Minutes + 1
         }
     }
+
+    /// <summary>
+    /// Extension class to add functionality to <see cref="DateTime"/> and <see cref="DateTimeOffset"/> so that use of <see cref="Duration"/> is similar to using <see cref="TimeSpan"/>
+    /// </summary>
+    public static class DateTimeHelper
```
Comment (thomaslevesque) [inline src/Iso8601DurationHelper/Duration.cs]: Could you please rename it to `DateTimeDurationExtensions`?

File: src/Iso8601DurationHelper/Duration.cs [hunk 2]
```diff
@@ -678,4 +678,32 @@ private enum DurationComponent
...
         }
     }
+
+    /// <summary>
+    /// Extension class to add functionality to <see cref="DateTime"/> and <see cref="DateTimeOffset"/> so that use of <see cref="Duration"/> is similar to using <see cref="TimeSpan"/>
+    /// </summary>
+    public static class DateTimeHelper 
+    {
+        /// <summary>
+        /// Adds a <see cref="Duration"/> value to a <see cref="DateTime"/> instant
```
Comment (thomaslevesque) [inline src/Iso8601DurationHelper/Duration.cs]: Why "instant"? Just "Adds a Duration value to a DateTime" would be fine, I think. (with the `<see>` tags you already included)

File: src/Iso8601DurationHelper/Duration.cs [hunk 3]
```diff
@@ -678,4 +678,32 @@ private enum DurationComponent
...
+        /// </summary>
+        /// <param name="date">The date to add to</param>
+        /// <param name="duration">The duration to add</param>
+        /// <returns></returns>
```
Comment (thomaslevesque) [inline src/Iso8601DurationHelper/Duration.cs]: Please also add a description of what it returns (maybe something similar to [`DateTime.Add(TimeSpan)`](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.add?view=net-7.0#returns))

**Review by thomaslevesque — CHANGES_REQUESTED** [review]
Hi @preeder,
Thanks for the PR!
I have a few minor remarks regarding the docs comment, otherwise this is great. If you can address the remarks, I'll merge and publish a new release ASAP.

File: src/Iso8601DurationHelper/Duration.cs (hunk 1 above)
Comment (preeder) [reply src/Iso8601DurationHelper/Duration.cs]: Done

File: src/Iso8601DurationHelper/Duration.cs (hunk 2 above)
Comment (preeder) [reply src/Iso8601DurationHelper/Duration.cs]: Done

File: src/Iso8601DurationHelper/Duration.cs (hunk 3 above)
Comment (preeder) [reply src/Iso8601DurationHelper/Duration.cs]: Done.  Very similar to DateTime[Offset].Add(TimeSpan).  Considered changing "... represented by **this** instance...", but that matches usage.  While one _could_ call an extension method directly, generally everyone uses it like the extension, right?

Comment (thomaslevesque) [reply src/Iso8601DurationHelper/Duration.cs]: Pretty much, yes. Thanks for the change!

**Review by thomaslevesque — APPROVED** [review]
Thanks again, @preeder!

**Comment by thomaslevesque** [comment]
I just published the new release to NuGet, it should be visible in a few minutes.

**Pull request merged**